compose/desktop/desktop/samples/src/jvmMain/kotlin/androidx/compose/desktop/examples/mouseclicks/Main.jvm.kt

```diff
@@ -0,0 +1,196 @@
+                            keyboardModifiers = { isAltPressed },
+                            onLongClick = {
+                                println("Gray: Alt + Long Right Click")
+                            }
+                        ) {
+                            println("Gray: Alt + Right Click")
+                        }
+                        .onClick(enabled = enabled, interactionSource = interactionSource1) {
+                            println("Gray: Left Click")
+                        }.indication(interactionSource1, LocalIndication.current)
+                    ) {
+                        Column {
+                            Text("Left Click, Alt + Right Click, Alt + Right LongClick")
+
+                            val interactionSource2 = remember { MutableInteractionSource() }
+
+                            Box(modifier = Modifier.padding(40.dp).size(100.dp).background(Color.Red)
+                                .onClick(
+                                    enabled = enabled,
+                                    interactionSource = interactionSource2,
+                                    matcher = PointerMatcher.mouse(PointerButton.Primary),
+                                    keyboardModifiers = { this.isShiftPressed },
+                                    onDoubleClick = {
+                                        println("Red: Shift + Left DoubleClick")
+                                    }
+                                ) {
+                                    println("Red: Shift + Left Click")
+                                }
+                                .onClick(
+                                    enabled = enabled,
+                                    interactionSource = interactionSource2,
+                                    matcher = PointerMatcher.mouse(PointerButton.Secondary)
+                                ) {
+                                    println("Red: Right Click")
+                                }.indication(interactionSource2, LocalIndication.current)
+                            ) {
+                                Text("Right Click, Shift + Left Click, Shift + Left DoubleClick")
+                            }
+                        }
+
+                    }
+                }
+
+                var isDragging by remember { mutableStateOf(false) }
+                val isDraggingCtrlPressed = derivedStateOf {  isDragging && isCtrlPressed }
+
+                println("Recomposing now")
+
+                Column(modifier = Modifier.padding(25.dp)) {
+                    Text("onDrag")
+                    var offset1 by remember { mutableStateOf(Offset.Zero) }
+                    Box(modifier = Modifier.offset { IntOffset(offset1.x.toInt(), offset1.y.toInt()) }
+                        .size(100.dp).background(Color.Blue).pointerInput(Unit) {
+                            detectDragGestures(
+                                matcher = PointerMatcher.mouse(PointerButton.Secondary),
```

Comment:
When I perform a drag on Mac with two fingers on touchpad, then release one finger, I still perform a drag. Is we have a bug here, or mac continues to treat a single finger as the right button?

Reply:
Same on my Mac. I don't think it's a bug (at least not on our side), as we simply receive Secondary button as pressed even though only one finger interacts with TouchPad.